internal/command/deploy/deploy.go

```diff
@@ -247,11 +266,41 @@ func useMachines(appConfig *app.Config, appCompact *api.AppCompact, args DeployW
 	}
 }
 
+// gets an app's name, whether its for apps v1 or v2
+func determineAppName(ctx context.Context) string {
+	appNameFromContext := app.NameFromContext(ctx)
+
+	if appNameFromContext == "" {
+		appNameFromContext = appv2.NameFromContext(ctx)
+
+	}
+
+	return appNameFromContext
+}
+
 // determineAppConfig fetches the app config from a local file, or in its absence, from the API
 func determineAppConfig(ctx context.Context) (cfg *app.Config, err error) {
 	tb := render.NewTextBlock(ctx, "Verifying app config")
 	client := client.FromContext(ctx).API()
-	appNameFromContext := app.NameFromContext(ctx)
+
+	appNameFromContext := determineAppName(ctx)
+	appCompact, err := client.GetAppCompact(ctx, appNameFromContext)
+
+	if err != nil {
+		return nil, err
+
+	}
+
+	if appCompact.PlatformVersion == appv2.AppsV2Platform {
+		return nil, wrongAppVersionErr
+
```

Comment:
I'm starting to believe this is your code editor or a anxious ENTER key 😁

Reply:
Honestly a reflex from writing Rust and C code, definitely not in line w the code base's code style. I'll clean these up where I can find them.